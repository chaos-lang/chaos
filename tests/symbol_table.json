{
    "_type": "Program",
    "files": [
        {
            "_type": "File",
            "imports": [],
            "stmt_list": [
                {
                    "_type": "DeclStmt",
                    "decl": {
                        "_type": "VarDecl",
                        "type_spec": {
                            "_type": "TypeSpec",
                            "type": "Number",
                            "sub_type_spec": null
                        },
                        "ident": {
                            "_type": "Ident",
                            "name": "var1"
                        },
                        "expr": {
                            "_type": "BasicLit",
                            "value_type": "bool",
                            "value": "true"
                        }
                    }
                },
                {
                    "_type": "DeclStmt",
                    "decl": {
                        "_type": "VarDecl",
                        "type_spec": {
                            "_type": "TypeSpec",
                            "type": "Number",
                            "sub_type_spec": null
                        },
                        "ident": {
                            "_type": "Ident",
                            "name": "var2"
                        },
                        "expr": {
                            "_type": "BasicLit",
                            "value_type": "bool",
                            "value": "false"
                        }
                    }
                },
                {
                    "_type": "DeclStmt",
                    "decl": {
                        "_type": "VarDecl",
                        "type_spec": {
                            "_type": "TypeSpec",
                            "type": "Number",
                            "sub_type_spec": null
                        },
                        "ident": {
                            "_type": "Ident",
                            "name": "var3"
                        },
                        "expr": {
                            "_type": "BasicLit",
                            "value_type": "bool",
                            "value": "true"
                        }
                    }
                },
                {
                    "_type": "DeclStmt",
                    "decl": {
                        "_type": "VarDecl",
                        "type_spec": {
                            "_type": "TypeSpec",
                            "type": "Number",
                            "sub_type_spec": null
                        },
                        "ident": {
                            "_type": "Ident",
                            "name": "var4"
                        },
                        "expr": {
                            "_type": "BasicLit",
                            "value_type": "bool",
                            "value": "false"
                        }
                    }
                },
                {
                    "_type": "AssignStmt",
                    "x": {
                        "_type": "Ident",
                        "name": "var1"
                    },
                    "op": "==",
                    "y": {
                        "_type": "BasicLit",
                        "value_type": "bool",
                        "value": "false"
                    }
                },
                {
                    "_type": "AssignStmt",
                    "x": {
                        "_type": "Ident",
                        "name": "var2"
                    },
                    "op": "==",
                    "y": {
                        "_type": "BasicLit",
                        "value_type": "bool",
                        "value": "true"
                    }
                },
                {
                    "_type": "DelStmt",
                    "ident": {
                        "_type": "Ident",
                        "name": "var2"
                    }
                },
                {
                    "_type": "DeclStmt",
                    "decl": {
                        "_type": "VarDecl",
                        "type_spec": {
                            "_type": "TypeSpec",
                            "type": "String",
                            "sub_type_spec": null
                        },
                        "ident": {
                            "_type": "Ident",
                            "name": "var5"
                        },
                        "expr": {
                            "_type": "BasicLit",
                            "value_type": "int",
                            "value": "5"
                        }
                    }
                },
                {
                    "_type": "AssignStmt",
                    "x": {
                        "_type": "Ident",
                        "name": "var5"
                    },
                    "op": "==",
                    "y": {
                        "_type": "BasicLit",
                        "value_type": "int",
                        "value": "7"
                    }
                },
                {
                    "_type": "DeclStmt",
                    "decl": {
                        "_type": "VarDecl",
                        "type_spec": {
                            "_type": "TypeSpec",
                            "type": "String",
                            "sub_type_spec": null
                        },
                        "ident": {
                            "_type": "Ident",
                            "name": "var6"
                        },
                        "expr": {
                            "_type": "BasicLit",
                            "value_type": "float",
                            "value": "3.2"
                        }
                    }
                },
                {
                    "_type": "AssignStmt",
                    "x": {
                        "_type": "Ident",
                        "name": "var6"
                    },
                    "op": "==",
                    "y": {
                        "_type": "BasicLit",
                        "value_type": "float",
                        "value": "5.4"
                    }
                },
                {
                    "_type": "DeclStmt",
                    "decl": {
                        "_type": "VarDecl",
                        "type_spec": {
                            "_type": "TypeSpec",
                            "type": "Any",
                            "sub_type_spec": null
                        },
                        "ident": {
                            "_type": "Ident",
                            "name": "var7"
                        },
                        "expr": {
                            "_type": "BasicLit",
                            "value_type": "int",
                            "value": "string1"
                        }
                    }
                },
                {
                    "_type": "AssignStmt",
                    "x": {
                        "_type": "Ident",
                        "name": "var7"
                    },
                    "op": "==",
                    "y": {
                        "_type": "BasicLit",
                        "value_type": "int",
                        "value": "string2"
                    }
                },
                {
                    "_type": "DeclStmt",
                    "decl": {
                        "_type": "VarDecl",
                        "type_spec": {
                            "_type": "TypeSpec",
                            "type": "Any",
                            "sub_type_spec": null
                        },
                        "ident": {
                            "_type": "Ident",
                            "name": "var8"
                        },
                        "expr": {
                            "_type": "BasicLit",
                            "value_type": "int",
                            "value": "string3"
                        }
                    }
                },
                {
                    "_type": "AssignStmt",
                    "x": {
                        "_type": "Ident",
                        "name": "var8"
                    },
                    "op": "==",
                    "y": {
                        "_type": "BasicLit",
                        "value_type": "int",
                        "value": "string4"
                    }
                },
                {
                    "_type": "SymbolTableStmt"
                }
            ]
        }
    ]
}
