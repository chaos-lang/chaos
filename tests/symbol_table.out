[start] =>
	{id: 0, name: var1, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 0, 2nd_type: 0, value_type: 0, role: 0, param_of: } =>
	{id: 2, name: var3, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 0, 2nd_type: 0, value_type: 0, role: 0, param_of: } =>
	{id: 3, name: var4, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 0, 2nd_type: 0, value_type: 0, role: 0, param_of: } =>
	{id: 4, name: var5, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 1, 2nd_type: 0, value_type: 1, role: 0, param_of: } =>
	{id: 5, name: var6, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 1, 2nd_type: 0, value_type: 2, role: 0, param_of: } =>
	{id: 6, name: var7, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{id: 7, name: var8, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
[end]
